src/aspire/volume/__init__.py

```diff
@@ -253,13 +254,82 @@ def downsample(self, szout, mask=None):
     def shift(self):
         raise NotImplementedError
 
-    def rotate(self):
-        raise NotImplementedError
+    def rotate(self, vol_idx, rot_matrices, nyquist=True):
+        """
+        Using the stack of rot_matrices,
+        rotate Volume[vol_idx].
+
+        :param vol_idx: Volume index
+        :param rot_matrices: Stack of rotations. Rotation or ndarray instance.
+        :return: `Volume` instance.
+        """
+
+        # If we are an ASPIRE Rotation, get the numpy representation.
+        if isinstance(rot_matrices, Rotation):
+            rot_matrices = rot_matrices.matrices()
+
+        if rot_matrices.dtype != self.dtype:
+            logger.warning(
+                f"{self.__class__.__name__}"
+                f" rot_matrices.dtype {rot_matrices.dtype}"
+                f" != self.dtype {self.dtype}."
+                " In the future this will raise an error."
+            )
+
+        data = self[vol_idx].T
+
+        pts_rot = rotated_grids_3d(self.resolution, rot_matrices)
+
+        vol_f = nufft(data, pts_rot)
+
+        vol_f = vol_f.reshape(-1, self.resolution, self.resolution, self.resolution)
+
+        # If resolution is even, we zero out the nyquist frequency by default.
+        if self.resolution % 2 == 0 and nyquist is True:
```

Comment:
We might want to add a little more to this comment so it is less confusing in the future.